cppcache/src/TcrEndpoint.cpp

```diff
@@ -504,14 +504,15 @@ void TcrEndpoint::pingServer(ThinClientPoolDM* poolDM) {
   }
 
   if (!m_msgSent && !m_pingSent) {
-    TcrMessagePing* pingMsg = TcrMessage::getPingMessage(m_cacheImpl);
+    TcrMessagePing pingMsg(new DataOutput(m_cacheImpl->createDataOutput()),
```

Comment:
Can you copy my PR where I modified the constructor to take a `std::unique_ptr<DataOutput>` to avoid any leaking potentials in allocating here and failing to take ownership in the constructor. Also makes it more clean who owns it.